pkg/querier/tenantfederation/merge_queryable.go

```diff
@@ -139,8 +139,19 @@ type mergeQuerier struct {
 func (m *mergeQuerier) LabelValues(name string, matchers ...*labels.Matcher) ([]string, storage.Warnings, error) {
 	log, _ := spanlogger.New(m.ctx, "mergeQuerier.LabelValues")
 	defer log.Span.Finish()
+
+	matchedValues, filteredMatchers := filterValuesByMatchers(m.idLabelName, m.ids, matchers...)
+	var labelValues []string
+	var ids []int
+	for labelPos := range m.ids {
```

Comment:
I think it would be way easier to pass the map `matchedValues` to a method `mergeDistinctStringSliceWithLabelValues` (with a nil map returning everything).